src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,39 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private static final int TIMEOUT = 5;
 
+    private final CopyOnWriteArrayList<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
+
+    public void registerListener(EventListener listener) {
+        listeners.addIfAbsent(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
     }
```

Comment:
Submitting tasks to executorService within a loop can potentially lead to a large number of tasks being queued simultaneously. Ensure that the system can handle the load or consider using a different approach to limit the number of concurrent tasks.